promgen/static/js/promgen.vue.js

```diff
@@ -92,27 +92,27 @@ var app = new Vue({
             let this_ = this;
             fetch('/proxy/v1/silences')
                 .then(response => response.json())
-                .then(function (silences) {
-                    this_.globalSilences = silences.data.sort(silence => silence.startsAt);
+                .then(function (response) {
+                    let silences = response.data.sort(silence => silence.startsAt);
 
                     // Pull out the matchers and do a simpler label map
-                    // To make other code easier
-                    for (var i in this_.globalSilences) {
-                        var silence = this_.globalSilences[i];
-                        silence.labels = {}
-                        for (var m in silence.matchers) {
-                            let matcher = silence.matchers[m]
-                            silence.labels[matcher.name] = matcher.value
+                    // to make other code easier
+                    for (let silence of silences) {
+                        silence.labels = {};
+                        for (let matcher of silence.matchers) {
+                            silence.labels[matcher.name] = matcher.value;
                         }
                     }
+
+                    this_.globalSilences = silences;
```

Comment:
Previous logic dynamically added properties to object observed by Vue, which can cause reactivity issues. This is solved by using an intermediate variable before assigning to `globalSilences`.